src/liboslexec/shadeimage.cpp

```diff
@@ -32,8 +32,13 @@ shade_image (ShadingSystem &shadingsys, ShaderGroup &group,
     if (! roi.defined())
         roi = buf.roi();
     if (buf.spec().format != TypeDesc::FLOAT) {
-        buf.error ("Cannot OSL::shade_image() into a %f buffer, float is required",
-                   buf.spec().format);
+#if OIIO_VERSION >= 20200
+        buf.errorfmt("Cannot OSL::shade_image() into a {} buffer, float is required",
+                     buf.spec().format);
```

Comment:
I think the OIIO_VERSION # is wrong.
https://github.com/OpenImageIO/oiio/blob/RB-2.2.3/src/include/OpenImageIO/imagebuf.h
As OIIOv2.2.3 doesn't contain errorfmt.
And my local build against 2.2.3 fails.

Evidence is RB-2.3.1 does contain it, suggest updating the OIIO_VERSION >= 20300

version

Reply:
Thanks, I think you are right, I will update. It was probably a typo.

Reply:
So, the current release is 2.2.12.  The FIRST public supported release of OIIO 2.2 was 2.2.6. So you are actually using some kind of development pre-release of OIIO.